runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/DefaultJobBundleFactory.java

```diff
@@ -374,17 +374,21 @@ ServerInfo getServerInfo() {
       return serverInfo;
     }
 
-    @Override
-    public void close() throws Exception {
+    public void close() {
       try (AutoCloseable envCloser = environment) {
         // Wrap resources in try-with-resources to ensure all are cleaned up.
```

Comment:
Actually, this comment was completely ignored. Adding all the `AutoCloseable`s in a single try block will call `close()` on **all** the resources and construct an exception which includes **all** the exceptions.